core/src/main/clojure/xtdb/expression.clj

```diff
@@ -240,6 +240,10 @@
 (defmethod codegen-cast [:num :utf8] [_]
   {:return-type :utf8, :->call-code #(do `(resolve-utf8-buf (str ~@%)))})
 
+(defmethod codegen-cast [:utf8 :uuid] [_]
+  {:return-type :uuid, :->call-code #(do `(util/uuid->byte-buffer
```

Comment:
@mpisanko Was just relooking at this. I think you wan to return a UUID here.

Reply:
looks like that - but having raised the inner form - a test fails (https://github.com/xtdb/xtdb/blob/main/src/test/clojure/xtdb/pgwire_test.clj#L1726) - so i guess it's allright - or the test is wrong?

Reply:
Yes, forget about this comment. I think we are dealing with multiple types (including UUID) in byte buffer form in the EE.